### PR TITLE
Fix bug that prevent users from using `guide-label` style to customize header labels

### PR DESCRIPTION
**Pull request by kanitw**
Fix https://github.com/vega/vega-lite/issues/3204

### CODE REVIEW

File: src/compile/layout/header.ts [hunk 1]
```diff
@@ -96,13 +96,11 @@ export function getHeaderGroup(model: Model, channel: HeaderChannel, headerType:
...
         encode: {
           update: {
-            fontWeight: {value: 'normal'},
-            angle: {value: 0},
-            fontSize: {value: 10}, // default label font size
             ... (channel === 'row' ? {
               align: {value: 'right'},
               baseline: {value: 'middle'}
             } : {})
+            // TODO: apply label* (e.g, labelAlign, labelBaseline) here
```
Comment (domoritz) [inline src/compile/layout/header.ts]: Is there an issue for this?

**Comment by domoritz** [comment]
Some tests are failing

screen shot 2017-11-22 at 08 58 31

**Comment by kanitw** [comment]
Hmm, this is weird.  I wonder if `keys(update).length` behave differently for linux.

**Comment by kanitw** [comment]
Should be good now.  The lower diff hit is actually beyond the scope of this PR.

**Pull request merged**